[2024-12-22T13:42:25.905+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-12-22T13:42:25.997+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: docker_dag.Notebook_job manual__2024-12-22T13:42:17.255482+00:00 [queued]>
[2024-12-22T13:42:26.022+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: docker_dag.Notebook_job manual__2024-12-22T13:42:17.255482+00:00 [queued]>
[2024-12-22T13:42:26.024+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2024-12-22T13:42:26.079+0000] {taskinstance.py:2889} INFO - Executing <Task(DockerOperator): Notebook_job> on 2024-12-22 13:42:17.255482+00:00
[2024-12-22T13:42:26.124+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=203) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-12-22T13:42:26.131+0000] {standard_task_runner.py:72} INFO - Started process 204 to run task
[2024-12-22T13:42:26.138+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'docker_dag', 'Notebook_job', 'manual__2024-12-22T13:42:17.255482+00:00', '--job-id', '151', '--raw', '--subdir', 'DAGS_FOLDER/A_dag_test.py', '--cfg-path', '/tmp/tmpx93daf5h']
[2024-12-22T13:42:26.141+0000] {standard_task_runner.py:105} INFO - Job 151: Subtask Notebook_job
[2024-12-22T13:42:26.273+0000] {task_command.py:467} INFO - Running <TaskInstance: docker_dag.Notebook_job manual__2024-12-22T13:42:17.255482+00:00 [running]> on host 49ab4fbc6be0
[2024-12-22T13:42:26.437+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='docker_dag' AIRFLOW_CTX_TASK_ID='Notebook_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-22T13:42:17.255482+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-12-22T13:42:17.255482+00:00'
[2024-12-22T13:42:26.439+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2024-12-22T13:42:26.439+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2024-12-22T13:42:26.440+0000] {logging_mixin.py:190} INFO - Current task name:Notebook_job state:running start_date:2024-12-22 13:42:25.998165+00:00
[2024-12-22T13:42:26.440+0000] {logging_mixin.py:190} INFO - Dag name:docker_dag and current dag run status:running
[2024-12-22T13:42:26.440+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-12-22T13:42:26.598+0000] {docker.py:379} INFO - Starting docker container from image spark-cluster:version-1.0.0
[2024-12-22T13:42:27.896+0000] {docker.py:73} INFO - Input Notebook:  /home/jovyan/Notebooks/NB_Calendar.ipynb
[2024-12-22T13:42:27.906+0000] {docker.py:73} INFO - Output Notebook: /home/jovyan/Notebooks/output-NB_Calendar.ipynb
[2024-12-22T13:42:27.962+0000] {docker.py:73} INFO - Executing:   0%|          | 0/14 [00:00<?, ?cell/s]
[2024-12-22T13:42:29.010+0000] {docker.py:73} INFO - Executing notebook with kernel: python3
[2024-12-22T13:42:29.890+0000] {docker.py:73} INFO - Executing:   7%|▋         | 1/14 [00:01<00:25,  1.93s/cell]
[2024-12-22T13:42:31.370+0000] {docker.py:73} INFO - :: loading settings :: url = jar:file:/usr/local/spark/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2024-12-22T13:42:31.431+0000] {docker.py:73} INFO - Ivy Default Cache set to: /home/jovyan/.ivy2/cache
[2024-12-22T13:42:31.432+0000] {docker.py:73} INFO - The jars for the packages stored in: /home/jovyan/.ivy2/jars
[2024-12-22T13:42:31.434+0000] {docker.py:73} INFO - io.delta#delta-spark_2.12 added as a dependency
[2024-12-22T13:42:31.435+0000] {docker.py:73} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-a3209d04-631d-4827-a9e5-1c83409058e4;1.0
[2024-12-22T13:42:31.435+0000] {docker.py:73} INFO - 	confs: [default]
[2024-12-22T13:42:33.504+0000] {docker.py:73} INFO - 	found io.delta#delta-spark_2.12;3.1.0 in central
[2024-12-22T13:42:34.156+0000] {docker.py:73} INFO - 	found io.delta#delta-storage;3.1.0 in central
[2024-12-22T13:42:38.458+0000] {docker.py:73} INFO - 	found org.antlr#antlr4-runtime;4.9.3 in central
[2024-12-22T13:42:38.712+0000] {docker.py:73} INFO - downloading https://repo1.maven.org/maven2/io/delta/delta-spark_2.12/3.1.0/delta-spark_2.12-3.1.0.jar ...
[2024-12-22T13:42:43.738+0000] {docker.py:73} INFO - 	[SUCCESSFUL ] io.delta#delta-spark_2.12;3.1.0!delta-spark_2.12.jar (5241ms)
[2024-12-22T13:42:44.012+0000] {docker.py:73} INFO - downloading https://repo1.maven.org/maven2/io/delta/delta-storage/3.1.0/delta-storage-3.1.0.jar ...
[2024-12-22T13:42:44.290+0000] {docker.py:73} INFO - 	[SUCCESSFUL ] io.delta#delta-storage;3.1.0!delta-storage.jar (550ms)
[2024-12-22T13:42:44.579+0000] {docker.py:73} INFO - downloading https://repo1.maven.org/maven2/org/antlr/antlr4-runtime/4.9.3/antlr4-runtime-4.9.3.jar ...
[2024-12-22T13:42:44.985+0000] {docker.py:73} INFO - 	[SUCCESSFUL ] org.antlr#antlr4-runtime;4.9.3!antlr4-runtime.jar (691ms)
[2024-12-22T13:42:44.987+0000] {docker.py:73} INFO - :: resolution report :: resolve 7051ms :: artifacts dl 6499ms
[2024-12-22T13:42:44.988+0000] {docker.py:73} INFO - 	:: modules in use:
[2024-12-22T13:42:44.989+0000] {docker.py:73} INFO - 	io.delta#delta-spark_2.12;3.1.0 from central in [default]
[2024-12-22T13:42:44.990+0000] {docker.py:73} INFO - 	io.delta#delta-storage;3.1.0 from central in [default]
[2024-12-22T13:42:44.990+0000] {docker.py:73} INFO - 	org.antlr#antlr4-runtime;4.9.3 from central in [default]
[2024-12-22T13:42:44.991+0000] {docker.py:73} INFO - 	---------------------------------------------------------------------
[2024-12-22T13:42:44.991+0000] {docker.py:73} INFO - 	|                  |            modules            ||   artifacts   |
[2024-12-22T13:42:44.992+0000] {docker.py:73} INFO - 	|       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[2024-12-22T13:42:44.992+0000] {docker.py:73} INFO - 	---------------------------------------------------------------------
[2024-12-22T13:42:44.993+0000] {docker.py:73} INFO - 	|      default     |   3   |   3   |   3   |   0   ||   3   |   3   |
[2024-12-22T13:42:44.993+0000] {docker.py:73} INFO - 	---------------------------------------------------------------------
[2024-12-22T13:42:45.009+0000] {docker.py:73} INFO - :: retrieving :: org.apache.spark#spark-submit-parent-a3209d04-631d-4827-a9e5-1c83409058e4
[2024-12-22T13:42:45.010+0000] {docker.py:73} INFO - 	confs: [default]
[2024-12-22T13:42:45.038+0000] {docker.py:73} INFO - 	3 artifacts copied, 0 already retrieved (5727kB/28ms)
[2024-12-22T13:42:45.377+0000] {docker.py:73} INFO - 24/12/22 13:42:45 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-12-22T13:42:45.519+0000] {docker.py:73} INFO - Setting default log level to "WARN".
[2024-12-22T13:42:45.519+0000] {docker.py:73} INFO - To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
[2024-12-22T13:42:46.332+0000] {docker.py:73} INFO - 24/12/22 13:42:46 WARN StandaloneSchedulerBackend: Dynamic allocation enabled without spark.executor.cores explicitly set, you may get more executors allocated than expected. It's recommended to set spark.executor.cores explicitly. Please check SPARK-30299 for more details.
[2024-12-22T13:42:46.398+0000] {docker.py:73} INFO - 24/12/22 13:42:46 WARN StandaloneAppClient$ClientEndpoint: Failed to connect to master spark-master:7077
[2024-12-22T13:42:46.398+0000] {docker.py:73} INFO - org.apache.spark.SparkException: Exception thrown in awaitResult: 
[2024-12-22T13:42:46.399+0000] {docker.py:73} INFO - 	at org.apache.spark.util.SparkThreadUtils$.awaitResult(SparkThreadUtils.scala:56)
[2024-12-22T13:42:46.400+0000] {docker.py:73} INFO - 	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:310)
[2024-12-22T13:42:46.400+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:75)
[2024-12-22T13:42:46.400+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:102)
[2024-12-22T13:42:46.400+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:110)
[2024-12-22T13:42:46.400+0000] {docker.py:73} INFO - 	at org.apache.spark.deploy.client.StandaloneAppClient$ClientEndpoint$$anon$1.run(StandaloneAppClient.scala:108)
[2024-12-22T13:42:46.400+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[2024-12-22T13:42:46.400+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[2024-12-22T13:42:46.401+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-22T13:42:46.401+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-22T13:42:46.401+0000] {docker.py:73} INFO - 	at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-22T13:42:46.401+0000] {docker.py:73} INFO - Caused by: java.io.IOException: Failed to connect to spark-master/<unresolved>:7077
[2024-12-22T13:42:46.401+0000] {docker.py:73} INFO - 	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:294)
[2024-12-22T13:42:46.401+0000] {docker.py:73} INFO - 	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:214)
[2024-12-22T13:42:46.401+0000] {docker.py:73} INFO - 	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:226)
[2024-12-22T13:42:46.401+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.netty.NettyRpcEnv.createClient(NettyRpcEnv.scala:204)
[2024-12-22T13:42:46.401+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:202)
[2024-12-22T13:42:46.402+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:198)
[2024-12-22T13:42:46.402+0000] {docker.py:73} INFO - 	... 4 more
[2024-12-22T13:42:46.402+0000] {docker.py:73} INFO - Caused by: java.net.UnknownHostException: spark-master
[2024-12-22T13:42:46.402+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress$CachedAddresses.get(InetAddress.java:801)
[2024-12-22T13:42:46.402+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1533)
[2024-12-22T13:42:46.402+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1385)
[2024-12-22T13:42:46.402+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1306)
[2024-12-22T13:42:46.402+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getByName(InetAddress.java:1256)
[2024-12-22T13:42:46.402+0000] {docker.py:73} INFO - 	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:156)
[2024-12-22T13:42:46.402+0000] {docker.py:73} INFO - 	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:153)
[2024-12-22T13:42:46.402+0000] {docker.py:73} INFO - 	at java.base/java.security.AccessController.doPrivileged(AccessController.java:569)
[2024-12-22T13:42:46.402+0000] {docker.py:73} INFO - 	at io.netty.util.internal.SocketUtils.addressByName(SocketUtils.java:153)
[2024-12-22T13:42:46.402+0000] {docker.py:73} INFO - 	at io.netty.resolver.DefaultNameResolver.doResolve(DefaultNameResolver.java:41)
[2024-12-22T13:42:46.403+0000] {docker.py:73} INFO - 	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:61)
[2024-12-22T13:42:46.403+0000] {docker.py:73} INFO - 	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:53)
[2024-12-22T13:42:46.403+0000] {docker.py:73} INFO - 	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:55)
[2024-12-22T13:42:46.403+0000] {docker.py:73} INFO - 	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:31)
[2024-12-22T13:42:46.403+0000] {docker.py:73} INFO - 	at io.netty.resolver.AbstractAddressResolver.resolve(AbstractAddressResolver.java:106)
[2024-12-22T13:42:46.403+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap.doResolveAndConnect0(Bootstrap.java:220)
[2024-12-22T13:42:46.403+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap.access$000(Bootstrap.java:46)
[2024-12-22T13:42:46.403+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:189)
[2024-12-22T13:42:46.403+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:175)
[2024-12-22T13:42:46.403+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
[2024-12-22T13:42:46.404+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557)
[2024-12-22T13:42:46.404+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
[2024-12-22T13:42:46.404+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636)
[2024-12-22T13:42:46.404+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:625)
[2024-12-22T13:42:46.404+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:105)
[2024-12-22T13:42:46.404+0000] {docker.py:73} INFO - 	at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84)
[2024-12-22T13:42:46.404+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetSuccess(AbstractChannel.java:990)
[2024-12-22T13:42:46.404+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:516)
[2024-12-22T13:42:46.404+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:429)
[2024-12-22T13:42:46.404+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:486)
[2024-12-22T13:42:46.404+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
[2024-12-22T13:42:46.404+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
[2024-12-22T13:42:46.404+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
[2024-12-22T13:42:46.405+0000] {docker.py:73} INFO - 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
[2024-12-22T13:42:46.405+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
[2024-12-22T13:42:46.405+0000] {docker.py:73} INFO - 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[2024-12-22T13:42:46.405+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
[2024-12-22T13:42:46.405+0000] {docker.py:73} INFO - 	... 1 more
[2024-12-22T13:43:06.420+0000] {docker.py:73} INFO - 24/12/22 13:43:06 WARN StandaloneAppClient$ClientEndpoint: Failed to connect to master spark-master:7077
[2024-12-22T13:43:06.424+0000] {docker.py:73} INFO - org.apache.spark.SparkException: Exception thrown in awaitResult: 
[2024-12-22T13:43:06.424+0000] {docker.py:73} INFO - 	at org.apache.spark.util.SparkThreadUtils$.awaitResult(SparkThreadUtils.scala:56)
[2024-12-22T13:43:06.424+0000] {docker.py:73} INFO - 	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:310)
[2024-12-22T13:43:06.424+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:75)
[2024-12-22T13:43:06.426+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:102)
[2024-12-22T13:43:06.427+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:110)
[2024-12-22T13:43:06.428+0000] {docker.py:73} INFO - 	at org.apache.spark.deploy.client.StandaloneAppClient$ClientEndpoint$$anon$1.run(StandaloneAppClient.scala:108)
[2024-12-22T13:43:06.431+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[2024-12-22T13:43:06.431+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[2024-12-22T13:43:06.431+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-22T13:43:06.432+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-22T13:43:06.435+0000] {docker.py:73} INFO - 	at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-22T13:43:06.436+0000] {docker.py:73} INFO - Caused by: java.io.IOException: Failed to connect to spark-master/<unresolved>:7077
[2024-12-22T13:43:06.436+0000] {docker.py:73} INFO - 	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:294)
[2024-12-22T13:43:06.436+0000] {docker.py:73} INFO - 	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:214)
[2024-12-22T13:43:06.438+0000] {docker.py:73} INFO - 	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:226)
[2024-12-22T13:43:06.439+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.netty.NettyRpcEnv.createClient(NettyRpcEnv.scala:204)
[2024-12-22T13:43:06.439+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:202)
[2024-12-22T13:43:06.439+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:198)
[2024-12-22T13:43:06.439+0000] {docker.py:73} INFO - 	... 4 more
[2024-12-22T13:43:06.439+0000] {docker.py:73} INFO - Caused by: java.net.UnknownHostException: spark-master
[2024-12-22T13:43:06.439+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress$CachedAddresses.get(InetAddress.java:801)
[2024-12-22T13:43:06.440+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1533)
[2024-12-22T13:43:06.440+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1385)
[2024-12-22T13:43:06.440+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1306)
[2024-12-22T13:43:06.440+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getByName(InetAddress.java:1256)
[2024-12-22T13:43:06.440+0000] {docker.py:73} INFO - 	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:156)
[2024-12-22T13:43:06.440+0000] {docker.py:73} INFO - 	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:153)
[2024-12-22T13:43:06.440+0000] {docker.py:73} INFO - 	at java.base/java.security.AccessController.doPrivileged(AccessController.java:569)
[2024-12-22T13:43:06.440+0000] {docker.py:73} INFO - 	at io.netty.util.internal.SocketUtils.addressByName(SocketUtils.java:153)
[2024-12-22T13:43:06.441+0000] {docker.py:73} INFO - 	at io.netty.resolver.DefaultNameResolver.doResolve(DefaultNameResolver.java:41)
[2024-12-22T13:43:06.441+0000] {docker.py:73} INFO - 	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:61)
[2024-12-22T13:43:06.441+0000] {docker.py:73} INFO - 	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:53)
[2024-12-22T13:43:06.441+0000] {docker.py:73} INFO - 	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:55)
[2024-12-22T13:43:06.441+0000] {docker.py:73} INFO - 	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:31)
[2024-12-22T13:43:06.441+0000] {docker.py:73} INFO - 	at io.netty.resolver.AbstractAddressResolver.resolve(AbstractAddressResolver.java:106)
[2024-12-22T13:43:06.441+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap.doResolveAndConnect0(Bootstrap.java:220)
[2024-12-22T13:43:06.441+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap.access$000(Bootstrap.java:46)
[2024-12-22T13:43:06.442+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:189)
[2024-12-22T13:43:06.442+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:175)
[2024-12-22T13:43:06.442+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
[2024-12-22T13:43:06.442+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557)
[2024-12-22T13:43:06.442+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
[2024-12-22T13:43:06.442+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636)
[2024-12-22T13:43:06.442+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:625)
[2024-12-22T13:43:06.443+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:105)
[2024-12-22T13:43:06.443+0000] {docker.py:73} INFO - 	at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84)
[2024-12-22T13:43:06.443+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetSuccess(AbstractChannel.java:990)
[2024-12-22T13:43:06.443+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:516)
[2024-12-22T13:43:06.443+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:429)
[2024-12-22T13:43:06.443+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:486)
[2024-12-22T13:43:06.443+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
[2024-12-22T13:43:06.443+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
[2024-12-22T13:43:06.444+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
[2024-12-22T13:43:06.444+0000] {docker.py:73} INFO - 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
[2024-12-22T13:43:06.444+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
[2024-12-22T13:43:06.444+0000] {docker.py:73} INFO - 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[2024-12-22T13:43:06.444+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
[2024-12-22T13:43:06.444+0000] {docker.py:73} INFO - 	... 1 more
[2024-12-22T13:43:26.462+0000] {docker.py:73} INFO - 24/12/22 13:43:26 WARN StandaloneAppClient$ClientEndpoint: Failed to connect to master spark-master:7077
[2024-12-22T13:43:26.468+0000] {docker.py:73} INFO - org.apache.spark.SparkException: Exception thrown in awaitResult: 
[2024-12-22T13:43:26.469+0000] {docker.py:73} INFO - 	at org.apache.spark.util.SparkThreadUtils$.awaitResult(SparkThreadUtils.scala:56)
[2024-12-22T13:43:26.469+0000] {docker.py:73} INFO - 	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:310)
[2024-12-22T13:43:26.469+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:75)
[2024-12-22T13:43:26.469+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:102)
[2024-12-22T13:43:26.469+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:110)
[2024-12-22T13:43:26.469+0000] {docker.py:73} INFO - 	at org.apache.spark.deploy.client.StandaloneAppClient$ClientEndpoint$$anon$1.run(StandaloneAppClient.scala:108)
[2024-12-22T13:43:26.469+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[2024-12-22T13:43:26.469+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[2024-12-22T13:43:26.469+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-22T13:43:26.469+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-22T13:43:26.470+0000] {docker.py:73} INFO - 	at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-22T13:43:26.470+0000] {docker.py:73} INFO - Caused by: java.io.IOException: Failed to connect to spark-master/<unresolved>:7077
[2024-12-22T13:43:26.470+0000] {docker.py:73} INFO - 	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:294)
[2024-12-22T13:43:26.470+0000] {docker.py:73} INFO - 	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:214)
[2024-12-22T13:43:26.470+0000] {docker.py:73} INFO - 	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:226)
[2024-12-22T13:43:26.470+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.netty.NettyRpcEnv.createClient(NettyRpcEnv.scala:204)
[2024-12-22T13:43:26.470+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:202)
[2024-12-22T13:43:26.470+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:198)
[2024-12-22T13:43:26.470+0000] {docker.py:73} INFO - 	... 4 more
[2024-12-22T13:43:26.470+0000] {docker.py:73} INFO - Caused by: java.net.UnknownHostException: spark-master
[2024-12-22T13:43:26.470+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress$CachedAddresses.get(InetAddress.java:801)
[2024-12-22T13:43:26.470+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1533)
[2024-12-22T13:43:26.471+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1385)
[2024-12-22T13:43:26.471+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1306)
[2024-12-22T13:43:26.471+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getByName(InetAddress.java:1256)
[2024-12-22T13:43:26.471+0000] {docker.py:73} INFO - 	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:156)
[2024-12-22T13:43:26.471+0000] {docker.py:73} INFO - 	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:153)
[2024-12-22T13:43:26.471+0000] {docker.py:73} INFO - 	at java.base/java.security.AccessController.doPrivileged(AccessController.java:569)
[2024-12-22T13:43:26.471+0000] {docker.py:73} INFO - 	at io.netty.util.internal.SocketUtils.addressByName(SocketUtils.java:153)
[2024-12-22T13:43:26.471+0000] {docker.py:73} INFO - 	at io.netty.resolver.DefaultNameResolver.doResolve(DefaultNameResolver.java:41)
[2024-12-22T13:43:26.471+0000] {docker.py:73} INFO - 	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:61)
[2024-12-22T13:43:26.471+0000] {docker.py:73} INFO - 	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:53)
[2024-12-22T13:43:26.471+0000] {docker.py:73} INFO - 	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:55)
[2024-12-22T13:43:26.471+0000] {docker.py:73} INFO - 	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:31)
[2024-12-22T13:43:26.471+0000] {docker.py:73} INFO - 	at io.netty.resolver.AbstractAddressResolver.resolve(AbstractAddressResolver.java:106)
[2024-12-22T13:43:26.472+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap.doResolveAndConnect0(Bootstrap.java:220)
[2024-12-22T13:43:26.472+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap.access$000(Bootstrap.java:46)
[2024-12-22T13:43:26.472+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:189)
[2024-12-22T13:43:26.472+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:175)
[2024-12-22T13:43:26.472+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
[2024-12-22T13:43:26.472+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557)
[2024-12-22T13:43:26.472+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
[2024-12-22T13:43:26.472+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636)
[2024-12-22T13:43:26.472+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:625)
[2024-12-22T13:43:26.472+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:105)
[2024-12-22T13:43:26.472+0000] {docker.py:73} INFO - 	at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84)
[2024-12-22T13:43:26.472+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetSuccess(AbstractChannel.java:990)
[2024-12-22T13:43:26.472+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:516)
[2024-12-22T13:43:26.473+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:429)
[2024-12-22T13:43:26.473+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:486)
[2024-12-22T13:43:26.473+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
[2024-12-22T13:43:26.473+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
[2024-12-22T13:43:26.473+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
[2024-12-22T13:43:26.473+0000] {docker.py:73} INFO - 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
[2024-12-22T13:43:26.473+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
[2024-12-22T13:43:26.473+0000] {docker.py:73} INFO - 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[2024-12-22T13:43:26.473+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
[2024-12-22T13:43:26.473+0000] {docker.py:73} INFO - 	... 1 more
[2024-12-22T13:43:46.417+0000] {docker.py:73} INFO - 24/12/22 13:43:46 ERROR StandaloneSchedulerBackend: Application has been killed. Reason: All masters are unresponsive! Giving up.
[2024-12-22T13:43:46.430+0000] {docker.py:73} INFO - 24/12/22 13:43:46 WARN StandaloneSchedulerBackend: Application ID is not initialized yet.
[2024-12-22T13:43:47.106+0000] {docker.py:73} INFO - 24/12/22 13:43:47 WARN Utils: spark.executor.instances less than spark.dynamicAllocation.minExecutors is invalid, ignoring its setting, please update your configs.
[2024-12-22T13:43:47.165+0000] {docker.py:73} INFO - 24/12/22 13:43:47 WARN StandaloneAppClient: Attempted to request executors before driver fully initialized.
[2024-12-22T13:43:47.848+0000] {docker.py:73} INFO - 24/12/22 13:43:47 WARN StandaloneAppClient$ClientEndpoint: Drop UnregisterApplication(null) because has not yet connected to master
[2024-12-22T13:43:48.647+0000] {docker.py:73} INFO - Executing:  21%|██▏       | 3/14 [01:20<05:26, 29.67s/cell]
[2024-12-22T13:43:48.674+0000] {docker.py:73} INFO - 24/12/22 13:43:48 WARN JavaUtils: Attempt to delete using native Unix OS command failed for path = /tmp/spark-63ec3308-0d8f-4f03-b8fa-dcf3734ad513. Falling back to Java IO way
[2024-12-22T13:43:48.677+0000] {docker.py:73} INFO - java.io.IOException: Failed to delete: /tmp/spark-63ec3308-0d8f-4f03-b8fa-dcf3734ad513
[2024-12-22T13:43:48.678+0000] {docker.py:73} INFO - 	at org.apache.spark.network.util.JavaUtils.deleteRecursivelyUsingUnixNative(JavaUtils.java:173)
[2024-12-22T13:43:48.679+0000] {docker.py:73} INFO - 	at org.apache.spark.network.util.JavaUtils.deleteRecursively(JavaUtils.java:109)
[2024-12-22T13:43:48.680+0000] {docker.py:73} INFO - 	at org.apache.spark.network.util.JavaUtils.deleteRecursively(JavaUtils.java:90)
[2024-12-22T13:43:48.681+0000] {docker.py:73} INFO - 	at org.apache.spark.util.SparkFileUtils.deleteRecursively(SparkFileUtils.scala:121)
[2024-12-22T13:43:48.681+0000] {docker.py:73} INFO - 	at org.apache.spark.util.SparkFileUtils.deleteRecursively$(SparkFileUtils.scala:120)
[2024-12-22T13:43:48.681+0000] {docker.py:73} INFO - 	at org.apache.spark.util.Utils$.deleteRecursively(Utils.scala:1126)
[2024-12-22T13:43:48.681+0000] {docker.py:73} INFO - 	at org.apache.spark.util.ShutdownHookManager$.$anonfun$new$4(ShutdownHookManager.scala:65)
[2024-12-22T13:43:48.681+0000] {docker.py:73} INFO - 	at org.apache.spark.util.ShutdownHookManager$.$anonfun$new$4$adapted(ShutdownHookManager.scala:62)
[2024-12-22T13:43:48.682+0000] {docker.py:73} INFO - 	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
[2024-12-22T13:43:48.682+0000] {docker.py:73} INFO - 	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
[2024-12-22T13:43:48.682+0000] {docker.py:73} INFO - 	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
[2024-12-22T13:43:48.682+0000] {docker.py:73} INFO - 	at org.apache.spark.util.ShutdownHookManager$.$anonfun$new$2(ShutdownHookManager.scala:62)
[2024-12-22T13:43:48.682+0000] {docker.py:73} INFO - 	at org.apache.spark.util.SparkShutdownHook.run(ShutdownHookManager.scala:214)
[2024-12-22T13:43:48.683+0000] {docker.py:73} INFO - 	at org.apache.spark.util.SparkShutdownHookManager.$anonfun$runAll$2(ShutdownHookManager.scala:188)
[2024-12-22T13:43:48.683+0000] {docker.py:73} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2024-12-22T13:43:48.683+0000] {docker.py:73} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:1928)
[2024-12-22T13:43:48.683+0000] {docker.py:73} INFO - 	at org.apache.spark.util.SparkShutdownHookManager.$anonfun$runAll$1(ShutdownHookManager.scala:188)
[2024-12-22T13:43:48.684+0000] {docker.py:73} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2024-12-22T13:43:48.684+0000] {docker.py:73} INFO - 	at scala.util.Try$.apply(Try.scala:213)
[2024-12-22T13:43:48.684+0000] {docker.py:73} INFO - 	at org.apache.spark.util.SparkShutdownHookManager.runAll(ShutdownHookManager.scala:188)
[2024-12-22T13:43:48.684+0000] {docker.py:73} INFO - 	at org.apache.spark.util.SparkShutdownHookManager$$anon$2.run(ShutdownHookManager.scala:178)
[2024-12-22T13:43:48.685+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[2024-12-22T13:43:48.685+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[2024-12-22T13:43:48.685+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-22T13:43:48.685+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-22T13:43:48.685+0000] {docker.py:73} INFO - 	at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-22T13:43:51.297+0000] {docker.py:73} INFO - Executing:  21%|██▏       | 3/14 [01:23<05:05, 27.77s/cell]
[2024-12-22T13:43:51.315+0000] {docker.py:73} INFO - Traceback (most recent call last):
[2024-12-22T13:43:51.315+0000] {docker.py:73} INFO -   File "/opt/conda/bin/papermill", line 8, in <module>
[2024-12-22T13:43:51.316+0000] {docker.py:73} INFO -     sys.exit(papermill())
[2024-12-22T13:43:51.317+0000] {docker.py:73} INFO -              ^^^^^^^^^^^
[2024-12-22T13:43:51.318+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
[2024-12-22T13:43:51.318+0000] {docker.py:73} INFO -     return self.main(*args, **kwargs)
[2024-12-22T13:43:51.319+0000] {docker.py:73} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-12-22T13:43:51.320+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/click/core.py", line 1078, in main
[2024-12-22T13:43:51.320+0000] {docker.py:73} INFO -     rv = self.invoke(ctx)
[2024-12-22T13:43:51.320+0000] {docker.py:73} INFO -          ^^^^^^^^^^^^^^^^
[2024-12-22T13:43:51.321+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
[2024-12-22T13:43:51.321+0000] {docker.py:73} INFO -     return ctx.invoke(self.callback, **ctx.params)
[2024-12-22T13:43:51.321+0000] {docker.py:73} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-12-22T13:43:51.321+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/click/core.py", line 783, in invoke
[2024-12-22T13:43:51.321+0000] {docker.py:73} INFO -     return __callback(*args, **kwargs)
[2024-12-22T13:43:51.322+0000] {docker.py:73} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-12-22T13:43:51.322+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/click/decorators.py", line 33, in new_func
[2024-12-22T13:43:51.322+0000] {docker.py:73} INFO -     return f(get_current_context(), *args, **kwargs)
[2024-12-22T13:43:51.322+0000] {docker.py:73} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-12-22T13:43:51.322+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/papermill/cli.py", line 235, in papermill
[2024-12-22T13:43:51.323+0000] {docker.py:73} INFO -     execute_notebook(
[2024-12-22T13:43:51.323+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/papermill/execute.py", line 131, in execute_notebook
[2024-12-22T13:43:51.323+0000] {docker.py:73} INFO -     raise_for_execution_errors(nb, output_path)
[2024-12-22T13:43:51.325+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/papermill/execute.py", line 251, in raise_for_execution_errors
[2024-12-22T13:43:51.326+0000] {docker.py:73} INFO -     raise error
[2024-12-22T13:43:51.327+0000] {docker.py:73} INFO - papermill.exceptions.PapermillExecutionError: 
[2024-12-22T13:43:51.328+0000] {docker.py:73} INFO - ---------------------------------------------------------------------------
[2024-12-22T13:43:51.329+0000] {docker.py:73} INFO - Exception encountered at "In [3]":
[2024-12-22T13:43:51.329+0000] {docker.py:73} INFO - ---------------------------------------------------------------------------
[2024-12-22T13:43:51.330+0000] {docker.py:73} INFO - Py4JJavaError                             Traceback (most recent call last)
[2024-12-22T13:43:51.330+0000] {docker.py:73} INFO - Cell In[3], line 2
[2024-12-22T13:43:51.331+0000] {docker.py:73} INFO -       1 # Configure Spark with Delta Lake (if needed)
[2024-12-22T13:43:51.332+0000] {docker.py:73} INFO - ----> 2 spark = configure_spark_with_delta_pip(builder).getOrCreate()
[2024-12-22T13:43:51.332+0000] {docker.py:73} INFO -       4 driver_version = sys.version_info
[2024-12-22T13:43:51.333+0000] {docker.py:73} INFO -       5 worker_version = spark.sparkContext.pythonExec
[2024-12-22T13:43:51.333+0000] {docker.py:73} INFO - 
[2024-12-22T13:43:51.333+0000] {docker.py:73} INFO - File /opt/conda/lib/python3.11/site-packages/pyspark/sql/session.py:500, in SparkSession.Builder.getOrCreate(self)
[2024-12-22T13:43:51.333+0000] {docker.py:73} INFO -     497     sc = SparkContext.getOrCreate(sparkConf)
[2024-12-22T13:43:51.333+0000] {docker.py:73} INFO -     498     # Do not update `SparkConf` for existing `SparkContext`, as it's shared
[2024-12-22T13:43:51.334+0000] {docker.py:73} INFO -     499     # by all sessions.
[2024-12-22T13:43:51.334+0000] {docker.py:73} INFO - --> 500     session = SparkSession(sc, options=self._options)
[2024-12-22T13:43:51.334+0000] {docker.py:73} INFO -     501 else:
[2024-12-22T13:43:51.334+0000] {docker.py:73} INFO -     502     getattr(
[2024-12-22T13:43:51.334+0000] {docker.py:73} INFO -     503         getattr(session._jvm, "SparkSession$"), "MODULE$"
[2024-12-22T13:43:51.335+0000] {docker.py:73} INFO -     504     ).applyModifiableSettings(session._jsparkSession, self._options)
[2024-12-22T13:43:51.335+0000] {docker.py:73} INFO - 
[2024-12-22T13:43:51.335+0000] {docker.py:73} INFO - File /opt/conda/lib/python3.11/site-packages/pyspark/sql/session.py:589, in SparkSession.__init__(self, sparkContext, jsparkSession, options)
[2024-12-22T13:43:51.335+0000] {docker.py:73} INFO -     585         getattr(getattr(self._jvm, "SparkSession$"), "MODULE$").applyModifiableSettings(
[2024-12-22T13:43:51.335+0000] {docker.py:73} INFO -     586             jsparkSession, options
[2024-12-22T13:43:51.335+0000] {docker.py:73} INFO -     587         )
[2024-12-22T13:43:51.336+0000] {docker.py:73} INFO -     588     else:
[2024-12-22T13:43:51.336+0000] {docker.py:73} INFO - --> 589         jsparkSession = self._jvm.SparkSession(self._jsc.sc(), options)
[2024-12-22T13:43:51.336+0000] {docker.py:73} INFO -     590 else:
[2024-12-22T13:43:51.336+0000] {docker.py:73} INFO -     591     getattr(getattr(self._jvm, "SparkSession$"), "MODULE$").applyModifiableSettings(
[2024-12-22T13:43:51.337+0000] {docker.py:73} INFO -     592         jsparkSession, options
[2024-12-22T13:43:51.337+0000] {docker.py:73} INFO -     593     )
[2024-12-22T13:43:51.337+0000] {docker.py:73} INFO - 
[2024-12-22T13:43:51.337+0000] {docker.py:73} INFO - File /opt/conda/lib/python3.11/site-packages/py4j/java_gateway.py:1587, in JavaClass.__call__(self, *args)
[2024-12-22T13:43:51.337+0000] {docker.py:73} INFO -    1581 command = proto.CONSTRUCTOR_COMMAND_NAME +\
[2024-12-22T13:43:51.338+0000] {docker.py:73} INFO -    1582     self._command_header +\
[2024-12-22T13:43:51.338+0000] {docker.py:73} INFO -    1583     args_command +\
[2024-12-22T13:43:51.338+0000] {docker.py:73} INFO -    1584     proto.END_COMMAND_PART
[2024-12-22T13:43:51.339+0000] {docker.py:73} INFO -    1586 answer = self._gateway_client.send_command(command)
[2024-12-22T13:43:51.339+0000] {docker.py:73} INFO - -> 1587 return_value = get_return_value(
[2024-12-22T13:43:51.339+0000] {docker.py:73} INFO -    1588     answer, self._gateway_client, None, self._fqn)
[2024-12-22T13:43:51.339+0000] {docker.py:73} INFO -    1590 for temp_arg in temp_args:
[2024-12-22T13:43:51.339+0000] {docker.py:73} INFO -    1591     if hasattr(temp_arg, "_detach"):
[2024-12-22T13:43:51.340+0000] {docker.py:73} INFO - 
[2024-12-22T13:43:51.340+0000] {docker.py:73} INFO - File /opt/conda/lib/python3.11/site-packages/py4j/protocol.py:326, in get_return_value(answer, gateway_client, target_id, name)
[2024-12-22T13:43:51.340+0000] {docker.py:73} INFO -     324 value = OUTPUT_CONVERTER[type](answer[2:], gateway_client)
[2024-12-22T13:43:51.340+0000] {docker.py:73} INFO -     325 if answer[1] == REFERENCE_TYPE:
[2024-12-22T13:43:51.340+0000] {docker.py:73} INFO - --> 326     raise Py4JJavaError(
[2024-12-22T13:43:51.341+0000] {docker.py:73} INFO -     327         "An error occurred while calling {0}{1}{2}.\n".
[2024-12-22T13:43:51.341+0000] {docker.py:73} INFO -     328         format(target_id, ".", name), value)
[2024-12-22T13:43:51.341+0000] {docker.py:73} INFO -     329 else:
[2024-12-22T13:43:51.342+0000] {docker.py:73} INFO -     330     raise Py4JError(
[2024-12-22T13:43:51.342+0000] {docker.py:73} INFO -     331         "An error occurred while calling {0}{1}{2}. Trace:\n{3}\n".
[2024-12-22T13:43:51.343+0000] {docker.py:73} INFO -     332         format(target_id, ".", name, value))
[2024-12-22T13:43:51.343+0000] {docker.py:73} INFO - 
[2024-12-22T13:43:51.343+0000] {docker.py:73} INFO - Py4JJavaError: An error occurred while calling None.org.apache.spark.sql.SparkSession.
[2024-12-22T13:43:51.344+0000] {docker.py:73} INFO - : java.lang.IllegalStateException: Cannot call methods on a stopped SparkContext.
[2024-12-22T13:43:51.345+0000] {docker.py:73} INFO - This stopped SparkContext was created at:
[2024-12-22T13:43:51.345+0000] {docker.py:73} INFO - 
[2024-12-22T13:43:51.345+0000] {docker.py:73} INFO - org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2024-12-22T13:43:51.346+0000] {docker.py:73} INFO - java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2024-12-22T13:43:51.346+0000] {docker.py:73} INFO - java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2024-12-22T13:43:51.346+0000] {docker.py:73} INFO - java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2024-12-22T13:43:51.347+0000] {docker.py:73} INFO - java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2024-12-22T13:43:51.347+0000] {docker.py:73} INFO - java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2024-12-22T13:43:51.347+0000] {docker.py:73} INFO - py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2024-12-22T13:43:51.347+0000] {docker.py:73} INFO - py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-12-22T13:43:51.347+0000] {docker.py:73} INFO - py4j.Gateway.invoke(Gateway.java:238)
[2024-12-22T13:43:51.348+0000] {docker.py:73} INFO - py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2024-12-22T13:43:51.348+0000] {docker.py:73} INFO - py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2024-12-22T13:43:51.348+0000] {docker.py:73} INFO - py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-12-22T13:43:51.348+0000] {docker.py:73} INFO - py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-12-22T13:43:51.348+0000] {docker.py:73} INFO - java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-22T13:43:51.349+0000] {docker.py:73} INFO - 
[2024-12-22T13:43:51.349+0000] {docker.py:73} INFO - The currently active SparkContext was created at:
[2024-12-22T13:43:51.349+0000] {docker.py:73} INFO - 
[2024-12-22T13:43:51.349+0000] {docker.py:73} INFO - (No active SparkContext.)
[2024-12-22T13:43:51.349+0000] {docker.py:73} INFO -          
[2024-12-22T13:43:51.350+0000] {docker.py:73} INFO - 	at org.apache.spark.SparkContext.assertNotStopped(SparkContext.scala:122)
[2024-12-22T13:43:51.350+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.SparkSession.<init>(SparkSession.scala:113)
[2024-12-22T13:43:51.350+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.SparkSession.<init>(SparkSession.scala:106)
[2024-12-22T13:43:51.351+0000] {docker.py:73} INFO - 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2024-12-22T13:43:51.351+0000] {docker.py:73} INFO - 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2024-12-22T13:43:51.351+0000] {docker.py:73} INFO - 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2024-12-22T13:43:51.351+0000] {docker.py:73} INFO - 	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2024-12-22T13:43:51.352+0000] {docker.py:73} INFO - 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2024-12-22T13:43:51.352+0000] {docker.py:73} INFO - 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2024-12-22T13:43:51.352+0000] {docker.py:73} INFO - 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-12-22T13:43:51.352+0000] {docker.py:73} INFO - 	at py4j.Gateway.invoke(Gateway.java:238)
[2024-12-22T13:43:51.353+0000] {docker.py:73} INFO - 	at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2024-12-22T13:43:51.353+0000] {docker.py:73} INFO - 	at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2024-12-22T13:43:51.353+0000] {docker.py:73} INFO - 	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-12-22T13:43:51.353+0000] {docker.py:73} INFO - 	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-12-22T13:43:51.354+0000] {docker.py:73} INFO - 	at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-22T13:43:52.505+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 519, in execute
    return self._run_image()
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 396, in _run_image
    return self._run_image_with_mounts(self.mounts, add_tmp_variable=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 453, in _run_image_with_mounts
    raise DockerContainerFailedException(f"Docker container failed: {result!r}", logs=log_lines)
airflow.providers.docker.exceptions.DockerContainerFailedException: Docker container failed: {'StatusCode': 1}
[2024-12-22T13:43:52.572+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2024-12-22T13:43:52.572+0000] {logging_mixin.py:190} INFO - Task start:2024-12-22 13:42:25.998165+00:00 end:2024-12-22 13:43:52.565444+00:00 duration:86.567279
[2024-12-22T13:43:52.573+0000] {logging_mixin.py:190} INFO - Task:<Task(DockerOperator): Notebook_job> dag:<DAG: docker_dag> dagrun:<DagRun docker_dag @ 2024-12-22 13:42:17.255482+00:00: manual__2024-12-22T13:42:17.255482+00:00, state:running, queued_at: 2024-12-22 13:42:17.304766+00:00. externally triggered: True>
[2024-12-22T13:43:52.574+0000] {logging_mixin.py:190} INFO - Failure caused by Docker container failed: {'StatusCode': 1}
[2024-12-22T13:43:52.575+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=docker_dag, task_id=Notebook_job, run_id=manual__2024-12-22T13:42:17.255482+00:00, execution_date=20241222T134217, start_date=20241222T134225, end_date=20241222T134352
[2024-12-22T13:43:52.610+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-12-22T13:43:52.613+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 151 for task Notebook_job (Docker container failed: {'StatusCode': 1}; 204)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 519, in execute
    return self._run_image()
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 396, in _run_image
    return self._run_image_with_mounts(self.mounts, add_tmp_variable=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 453, in _run_image_with_mounts
    raise DockerContainerFailedException(f"Docker container failed: {result!r}", logs=log_lines)
airflow.providers.docker.exceptions.DockerContainerFailedException: Docker container failed: {'StatusCode': 1}
[2024-12-22T13:43:52.655+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-12-22T13:43:52.757+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-12-22T13:43:52.767+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
